buildvertical.targets

```diff
@@ -0,0 +1,204 @@
+
+    <Message Importance="High" Text="$(MSBuildProjectFullPath) Adding $(_VerticalProject) KP:$(_KeepProject), SelectedTargetGroup: $(_SelectedTargetGroup), SelectedOSGroup: $(_SelectedOSGroup)" />
+    <ItemGroup Condition="'$(_KeepProject)' != 'true'">
+      <Project Remove="$(_VerticalProject)" />
+      <Project Condition="'$(_SelectedTargetGroup)' != '' and '$(_SelectedOSGroup)' != ''" Include="$(_VerticalProjectName)">  
+        <TargetGroup>$(_SelectedTargetGroup)</TargetGroup>
+        <OSGroup Condition="'$(_SelectedOSGroup)' != 'AnyOS'">$(_SelectedOSGroup)</OSGroup>
+        <Configuration Condition="'$(_Configuration)' != ''">$(_Configuration)</Configuration>
+      </Project>
+      <Project Condition="'$(_SelectedTargetGroup)' == ''" Include="$(_VerticalProjectName)">
+        <OmitReason>No applicable TargetGroup</OmitReason>
+      </Project>
+      <Project Condition="'$(_SelectedOSGroup)' == ''" Include="$(_VerticalProjectName)">
+        <OmitReason>No applicable OSGroup</OmitReason>
+      </Project>
+      <Project Condition="'$(_OmitReason)' != ''">
+        <OmitReason>$(_OmitReason)</OmitReason>
+      </Project>
+    </ItemGroup>
+    <ItemGroup Condition="'$(_KeepProject)' == 'true'">
+      <Project Remove="$(_VerticalProject)" />
+      <Project Include="$(_VerticalProjectName)" />
+    </ItemGroup>
+  </Target>
+
+  <!-- For the current project, target which determines compatibility for targets -->
+  <Target Name="FindBestConfiguration"
+          Returns="@(VerticalTargets)"
+          BeforeTargets="BeforeResolveReferences">
+    
+    <Message Importance="Low" Text="FindBestConfiguration: $(MSBuildProjectFullPath) - $(BuildConfigurationImportFile)" />
+    <MSBuild Targets="GetBuildConfigurations"
+             Projects="$(MSBuildProjectFullPath)">
+      <Output TaskParameter="TargetOutputs"
+              PropertyName="BuildConfigurations" />
+    </MSBuild>
+
+    <Message Text="$(MSBuildProjectFullPath) - BuildConfigurations: $(BuildConfigurations)" Importance="High" />
+
+    <Message Importance="Normal" Condition="'$(BuildConfigurations)' == ''" Text="Project '$(MSBuildProjectFullPath)' does not specify the BuildConfigurations property." />
+
+     <!-- Clear the VerticalTargets item -->
+    <ItemGroup><VerticalTargets Remove="@(VerticalTargets)" /></ItemGroup>
+    <FindBestConfiguration Properties="@(Property)"
+                           PropertyValues="@(PropertyValue)"
+                           BuildConfigurations="$(BuildConfigurations)"
+                           BuildConfiguration="$(BuildConfiguration)" 
+                           Condition="'$(BuildConfigurations)' != ''"
+                           ContinueOnError="true">
+      <Output TaskParameter="BestConfiguration" ItemName="VerticalTargets" />                          
+    </FindBestConfiguration>  
+
+    <ItemGroup Condition="'$(BuildConfigurations)' == ''">
+      <VerticalTargets Include="$(MSBuildProjectFullPath)">
```

Comment:
If BuildConfigurations is empty then lets set use the current BuildConfiguration as BestConfiguration.